pym/bob/input.py

```diff
@@ -1950,6 +1959,12 @@ def prepare(self, pathFormatter, inputEnv, sandboxEnabled, inputStates, inputSan
             varPrivate[key] = env.substitute(value, "privateEnvironment::"+key)
         env.update(varPrivate)
 
+        # apply meta variables
+        for (key, value) in self.__varMeta.items():
+            if key in env:
+                raise ParseError("Duplikate variable in metaVars and environment: " + key)
```

Comment:
The only warning I could think of is when you specify the same variable in `privateEnvironment` and `metaEnvironment`. The latter will always win, so it makes no sense. I don't think a warning should be triggered if the variable was inherited from somewhere else.